File: Java/1.java
// https://leetcode.com/problems/two-sum/
// Runtime: 1 ms, faster than 99.72% of Java online submissions for Two Sum.
// Memory Usage: 42.3 MB, less than 90.35% of Java online submissions for Two Sum.
// 04/14/2022 21:31
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> found = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int need = target - nums[i];
            if (found.containsKey(need)) {
                return new int[] {found.get(need), i};
            }
            found.put(nums[i], i);
        }
        return new int[] {-1, -1};
    }
}

File: Python/1.py
# https://leetcode.com/problems/two-sum
# Runtime: 125 ms, faster than 41.13% of Python3 online submissions for Two Sum.
# Memory Usage: 15.2 MB, less than 50.04% of Python3 online submissions for Two Sum.
# 05/05/2022 20:18
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        found: dict[int, int] = {}
        for i, num in enumerate(nums):
            comp: int = target - num
            if comp in found:
                return [found[comp], i]
            found[num] = i
        return [-1, -1]
